### PR TITLE
Split Collector from the cobra.Command.

### PR DESCRIPTION
**Pull request by bogdandrutu**
Updates https://github.com/open-telemetry/opentelemetry-collector/issues/3957

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: service/collector.go [hunk 1]
```diff
@@ -246,7 +201,11 @@ func (col *Collector) setupConfigurationComponents(ctx context.Context) error {
 	return nil
 }
 
-func (col *Collector) execute(ctx context.Context) error {
+// Run starts the collector according to the command and configuration
+// given by the user, and waits for it to complete.
+// Consecutive calls to Run are not allowed, Run shouldn't be called
+// once a collector is shut down.
+func (col *Collector) Run(ctx context.Context) error {
```
Comment (mx-psi) [inline service/collector.go:208]: Should we add a unit test calling `Run` without the command wrapper?

File: service/collector.go [hunk 2]
```diff
@@ -246,7 +201,11 @@ func (col *Collector) setupConfigurationComponents(ctx context.Context) error {
 	return nil
 }
 
-func (col *Collector) execute(ctx context.Context) error {
+// Run starts the collector according to the command and configuration
```
Comment (mx-psi) [inline service/collector.go:204]: It does not take the command into account anymore :)
```suggestion
// Run starts the collector according to the configuration
```

**Review by mx-psi — COMMENTED** [review]
Thanks so much for doing this 🙇  This looks better than my proposal on the issue (I didn't want to break things but this is better if it's okay to break things).

File: service/collector.go (hunk 1 above)
Comment (mx-psi) [reply service/collector.go:206]: Ah, the `TestCollector_StartAsGoRoutine` already does it. No need to add a new one then.

Comment (bogdandrutu) [reply service/collector.go:206]: I've just changed that to do it based on your feedback 😄

**Review by mx-psi — APPROVED** [review]

File: service/collector.go (hunk 1 above)
Comment (mx-psi) [reply service/collector.go:206]: hard to see with the force push haha, LGTM then :)

**Review by dashpole — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

File: service/command.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	rootCmd := &cobra.Command{
+		Use:          col.set.BuildInfo.Command,
+		Version:      col.set.BuildInfo.Version,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return col.Run(cmd.Context())
+		},
+	}
+
+	// TODO: coalesce this code and expose this information to other components.
```
Comment (codeboten) [inline service/command.go:40]: Is this TODO still needed? If so is there an issue open for it?

**Pull request merged**